Source/WebCore/platform/graphics/avfoundation/FormatDescriptionUtilities.cpp

```diff
@@ -214,30 +214,42 @@ std::optional<SpatialVideoMetadata> videoMetadataFromFormatDescription(CMFormatD
-    metadata.horizontalFOVDegrees = value / 1000.0;
-
-    auto baselineField = dynamic_cf_cast<CFNumberRef>(PAL::CMFormatDescriptionGetExtension(formatDescription, PAL::kCMFormatDescriptionExtension_StereoCameraBaseline));
-    if (!baselineField)
-        return { };
-    CFNumberGetValue(baselineField, kCFNumberSInt32Type, &value);
-    metadata.baseline = value;
-
-    auto disparityAdjustmentField = dynamic_cf_cast<CFNumberRef>(PAL::CMFormatDescriptionGetExtension(formatDescription, PAL::kCMFormatDescriptionExtension_HorizontalDisparityAdjustment));
-    if (!disparityAdjustmentField)
-        return { };
-    CFNumberGetValue(disparityAdjustmentField, kCFNumberSInt32Type, &value);
-    metadata.disparityAdjustment = value / 10000.0;
-
-    CMVideoDimensions dimensions = PAL::CMVideoFormatDescriptionGetDimensions(formatDescription);
-    metadata.size = { dimensions.width, dimensions.height };
-
-    return metadata;
+    auto checkForSpatialMetadata = [&]() -> std::optional<SpatialVideoMetadata> {
+        SInt32 value;
+        SpatialVideoMetadata metadata;
+        auto horizontalFieldOfView = dynamic_cf_cast<CFNumberRef>(PAL::CMFormatDescriptionGetExtension(formatDescription, PAL::kCMFormatDescriptionExtension_HorizontalFieldOfView));
+        if (!horizontalFieldOfView)
+            return { };
+        CFNumberGetValue(horizontalFieldOfView, kCFNumberSInt32Type, &value);
+        metadata.horizontalFOVDegrees = value / 1000.0;
+
+        auto baselineField = dynamic_cf_cast<CFNumberRef>(PAL::CMFormatDescriptionGetExtension(formatDescription, PAL::kCMFormatDescriptionExtension_StereoCameraBaseline));
+        if (!baselineField)
+            return { };
+        CFNumberGetValue(baselineField, kCFNumberSInt32Type, &value);
+        metadata.baseline = value;
+
+        auto disparityAdjustmentField = dynamic_cf_cast<CFNumberRef>(PAL::CMFormatDescriptionGetExtension(formatDescription, PAL::kCMFormatDescriptionExtension_HorizontalDisparityAdjustment));
+        if (!disparityAdjustmentField)
+            return { };
+        CFNumberGetValue(disparityAdjustmentField, kCFNumberSInt32Type, &value);
+        metadata.disparityAdjustment = value / 10000.0;
+
+        CMVideoDimensions dimensions = PAL::CMVideoFormatDescriptionGetDimensions(formatDescription);
+        metadata.size = { dimensions.width, dimensions.height };
+        return metadata;
+    };
+    if (auto spatialVideoMetadata = checkForSpatialMetadata())
+        return spatialVideoMetadata;
+
+    auto checkForImmersiveData = [&]() -> bool {
+        auto projectionKind = dynamic_cf_cast<CFStringRef>(PAL::CMFormatDescriptionGetExtension(formatDescription, CFSTR("ProjectionKind")));
+        return projectionKind && (CFStringCompare(projectionKind, CFSTR("Equirectangular"), 0) == kCFCompareEqualTo || CFStringCompare(projectionKind, CFSTR("HalfEquirectangular"), 0) == kCFCompareEqualTo || CFStringCompare(projectionKind, CFSTR("ParametricImmersive"), 0) == kCFCompareEqualTo);
```

Comment:
You should add `kCMFormatDescriptionProjectionKind_Equirectangular` and `kCMFormatDescriptionProjectionKind_HalfEquirectangular` to CoreMediaSoftLink

Reply:
could have sworn they weren't in the public SDK when I looked last !